GDK Update Compatibility + Custom DLL URL Option

ImTheGoobie opened the pull request:
The launcher now can reliably launch the game, however injection is not 100% reliable, works most of the time for me.
Version detection for left click launch is janky, definitely needs fixing.

Commit:
Fixed launching the game and partially reliable injection, needs some work though

Commit:
Removed threading and unused collections

Commit:
Update LatiteFolder path and removed RoamingStateDirectory

Commit:
Added option for custom dll url

ImTheGoobie commented:
The custom DLL option is useful for when an in progress dll is available but not in nightly or stable. Might need a little change if not up to standards.

Plextora commented:
I'm pretty sure that `ApplyAppPackages` in `Injector.cs` can safely be removed, as the use of this method was some permission stuff relating to UWP

Plextora commented:
~~Right clicking the Launch button for selecting a local Custom DLL seems to be broken (doesn't inject) unless I press the minimize button on the Minecraft window~~

~~My guess for the reason is this is that the module count for Minecraft being fully loaded probably changed, breaking `Injector.WaitForModules` (I haven't validated this theory however, just a guess)~~ ~~Scratch that, dll injection doesn't work when left clicking the Launch button either, and it seems like the cause is in `Injector.WaitForMinecraft`. Not sure what's wrong with that function though~~ NEVERMIND ‼‼

Plextora commented on LatiteInjector/Utils/Updater.cs:
```diff
@@ -119,6 +119,9 @@ public static async Task<string> DownloadDll()
         string debugPdbPath = $"{Logging.LatiteFolder}\\LatiteDebug.pdb";
 
+        string customDllUrl = SettingsWindow.CustomDLLURL;
+        string customDllPath = $"{Logging.LatiteFolder}\\Custom_DLL.dll";
```
Custom DLLs downloaded via URL should be downloaded to the `Logging.InjectorFolder` directory, not the `Logging.LatiteFolder` directory.

Moreover, whenever using `Logging.LatiteFolder` there should be a check to make sure the folder is created, if not then create the folder in order to prevent an `System.IO.DirectoryNotFoundException` Exception.

We should also probably just delete `LatiteInjectorDataFolder` and use `Logging.InjectorFolder` codebase wide since its kinda redundant to have multiple variables leading to the same path.

Commit:
Remove ApplyAppPackages and PreventAppSuspension as these are no longer mandatory

ImTheGoobie commented:
> I'm pretty sure that `ApplyAppPackages` in `Injector.cs` can safely be removed, as the use of this method was some permission stuff relating to UWP

Seems to work without it just fine, removed it and cleaned up some previously commented code as this was also not required anymore.

Plextora commented:
LGTM 👍
